head/firmware/motor_hardware.c

```diff
@@ -5,6 +5,7 @@
 TIM_HandleTypeDef htim7;
 TIM_HandleTypeDef htim2;
 TIM_HandleTypeDef htim3;
+uint32_t enc_pulse_counter;
```

Comment:
Is this variable meant to be accessible outside of this module using `extern`? If not, can you please make it `static`

Reply:
yep, that variable is meant to be accessible outside of the module. Add this to motor_hardware.h as `extern enc_pulse_counter;` ?